### PR TITLE
reader: fix jgf properties

### PR DESCRIPTION
**Pull request by jameshcorbett**
Given this vertex in JGF, resource-query reports it has no properties.

```json
      {
        "id": "35",
        "metadata": {
          "type": "node",
          "basename": "node",
          "name": "compute-01",
          "id": 1,
          "uniq_id": 35,
          "rank": 0,
          "exclusive": true,
          "unit": "",
          "size": 1,
          "properties": ["pdebug"],
          "paths": {
            "containment": "/ElCapitan0/rack0/compute-01"
          }
        }
      }
```

```
resource-query> g /ElCapitan0/rack0/compute-01
No properties were found for /ElCapitan0/rack0/compute-01 (vtx's uniq_id=35).
```

Looking at the reader, I found that it expects properties to be given as a mapping with keys the name of the property. When I changed the vertex in JGF to the following, the property was recognized. It looks like the reader had been silently failing on the type.

```json

      {
        "id": "35",
        "metadata": {
          "type": "node",
          "basename": "node",
          "name": "compute-01",
          "id": 1,
          "uniq_id": 35,
          "rank": 0,
          "exclusive": true,
          "unit": "",
          "size": 1,
          "properties": {
            "pdebug": ""
          },
          "paths": {
            "containment": "/ElCapitan0/rack0/compute-01"
          }
        }
      },
```

```
resource-query> g /ElCapitan0/rack0/compute-01
pdebug=
```

### CODE REVIEW

**Comment by jameshcorbett** [comment]
@milroy this is ready for review whenever.

File: t/python/t10001-resourcegraph.py [hunk 1]
```diff
@@ -45,6 +45,17 @@
     },
```
Comment (milroy) [inline t/python/t10001-resourcegraph.py:45]: Typo in the commit message. I think you mean "there are no tests where the JGF writer handles properties correctly."

**Review by milroy — APPROVED** [review]
Apart from a typo in one commit message this PR LGTM!

File: t/python/t10001-resourcegraph.py (hunk 1 above)
Comment (jameshcorbett) [reply t/python/t10001-resourcegraph.py:45]: Good catch, thanks! Force-pushed with the correction.

**Comment by jameshcorbett** [comment]
Thanks! Setting MWP.

**Pull request merged**